### ISSUE TITLE
shouldNotThrow not working

### ISSUE THREAD

**Issue by svenjacobs** [MarkusAmshove/Kluent#128]
Kluent: 1.44
Kotlin: 1.3.11

`shouldNotThrow` isn't working at all for me. The following code does **not** fail:

```kotlin
val fn = {
    throw Exception("")
}

fn shouldNotThrow Exception::class
```

I think the problem is in [this](https://github.com/MarkusAmshove/Kluent/blob/master/jvm/src/main/kotlin/org/amshove/kluent/Exceptions.kt#L23) line. `isAnyException()` performs a comparison of `javaObjectType` of the current exception object with `AnyException.javaObjectType`. These types are never the same.

![image](https://user-images.githubusercontent.com/255313/50217163-4172f800-0388-11e9-88b7-bbb62403e129.png)

**Comment by MarkusAmshove** [comment]
You're right, that was the problem.

I've fixed it and released v1.45, you can try it out

**Comment by svenjacobs** [comment]
Yes, it was fixed with version 1.45. Thank you!
